packages/cog/src/cli/actions/action.cog.ts

```diff
@@ -11,6 +11,33 @@ import { buildCogForQuadKey, CogJob } from '../../cog/cog';
 import { buildWarpedVrt } from '../../cog/cog.vrt';
 import { getJobPath, makeTempFolder } from '../folder';
 
+export function extractYearFromName(name: string): number {
+    const re = /(?:^|\D)(\d{4})(?:$|\D)/g;
+
+    let year = 2000;
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        const t = +m[1];
+        if (t > year) year = t;
+    }
+
+    return year;
+}
+
+export function storeImage(job: CogJob): void {
+    const now = Date.now();
+    new TileMetadataTable().create({
+        id: job.id, // FIXME - should this be calculated by name?
```

Comment:
is job.id right?

Reply:
This `storeImage` looks to be getting called every time we process a cog, so for some imagery sets this will be called 100+ times.

I wonder if we want to run a quadkey `addToSet` `removeFromSet` action rather than create

Reply:
Should this be done in action.batch rather than action.cog?

Reply:
We could insert a record inside of `batch` then update it inside of `cog`.

Insert a record with `quadKeys: []` and then inside of cog we could do a `addToSet: quadkey :quadKeys` (not sure the exact syntax)

Reply:
What is `addToSet`?  wouldn't updating in action.cog still result in 100+ writes?

Reply:
100 writes are fine if you are actually modifying something.

In this PR, you are writing the same data 100s of time.

if we were doing a `addToArray` we would be updating what "quadKeys" have been processed

Examples on how to adjust an array with dynamodb updates.

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.SET.AddingListElements

Reply:
The problem with updating the `quadKeys` attribute is that the order in which the `gdal_translate` finishes is indeterminate and I can not therefore easily update the array in proper order. What is wrong with just creating the entire record in `action.batch` prior to job submission and not performing any updates in `action.cog`?

Reply:
The order doesnt actually matter, it will check each quadkey for intersections.

it was only ordered so that I could look at them nicely.

but either way is fine.